src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectManaged.cs

```diff
@@ -81,39 +77,25 @@ protected override void RemovePacketFromPendingList(PacketHandle packet)
 
         internal override void Dispose()
         {
-            SNIPacket packetHandle = _sniPacket;
             SNIHandle sessionHandle = _sessionHandle;
-            SNIPacket asyncAttnPacket = _sniAsyncAttnPacket;
 
-            _sniPacket = null;
             _sessionHandle = null;
-            _sniAsyncAttnPacket = null;
             _marsConnection = null;
 
             DisposeCounters();
 
-            if (null != sessionHandle || null != packetHandle)
+            if (sessionHandle != null)
             {
-                packetHandle?.Dispose();
```

Comment:
Where is the packetHandle being disposed in the new code?

Reply:
NVM, I didnt realize I am looking at TdsParserStateObjectManaged where you have changed the lifecycle of packet.

Reply:
I've tried to make sure that every packet is correctly released which means there are now Release calls scattered throughout the code. It isn't the end of the world is a packet is dropped without clearing but it'd rather it didn't happen.

As I mentioned my eventual intention is to pool the packets so that a small number are kept and reused for all connections as this will substantially reduce allocations.

Reply:
Yeah I understand. I didn't clarify. It seemed to me that I was read the TPSONative code and it seemed that the dispose logic which existed earlier, has been removed. Later I realized that this is managed TDSO where you are taking care of the cleanup in the SNI layer itself. 
Had the disposed been removed in native stack, there should have been a lot of mem leaks.